vochain/app.go

```diff
@@ -42,13 +42,14 @@ var (
 // BaseApplication reflects the ABCI application implementation.
 type BaseApplication struct {
 	State              *vstate.State
+	Istc               *ist.Controller
 	Service            service.Service
 	Node               *tmcli.Local
 	TransactionHandler *transaction.TransactionHandler
-	IsSynchronizing    func() bool
+	isSynchronizingFn  func() bool
 	// tendermint WaitSync() function is racy, we need to use a mutex in order to avoid
 	// data races when querying about the sync status of the blockchain.
-	isSyncLock sync.Mutex
+	isSynchronizing uint32
```

Comment:
Why not just use `atomic.Uint32`?

Reply:
Yes, I think they are both the same, but probably using the atomic version would be clearer. That way, just by reading the variable declaration you know that it is atomic, currently you have to look into the `IsSynchronizing()` method to know that.

Reply:
you could also simply export an https://pkg.go.dev/sync/atomic#Bool, removing the need for uint32, the Fn field, your own method, etc.